lib/wasi/src/runtime/resolver/types.rs

```diff
@@ -0,0 +1,197 @@
+use std::{collections::BTreeMap, fmt::Display, ops::Deref, path::PathBuf, str::FromStr};
+
+use crate::{
+    bin_factory::BinaryPackage,
+    http::HttpClient,
+    runtime::resolver::{cache::CacheConfig, InMemoryCache},
+};
+
+#[async_trait::async_trait]
+pub trait PackageResolver: std::fmt::Debug + Send + Sync {
+    /// Resolve a package, loading all dependencies.
+    async fn resolve_package(
```

Comment:
I think this is too limited.

There are different use cases that need access to different things:

A) The manifest for a package. This could be needed during dependency resolution, or during other resolution in different contexts. Manifests could be cached efficiently as well, (like cargo does with the new HTTP index access method) , or directly loaded from local webc files. 

B) The raw webc 

C) A loaded `WebC` for directly accessing the manifest, the volumes, etc

D) A materialized binary package

Reply:
Yeah, in the next PR I'll start fleshing the `PackageResolver` trait out a bit more.